src/token/src/core.rs

```diff
@@ -0,0 +1,355 @@
+use std::cell::RefCell;
+use std::future::Future;
+use std::pin::Pin;
+use std::rc::Rc;
+
+use candid::Principal;
+use ic_canister::{query, update, Canister};
+use ic_helpers::tokens::Tokens128;
+
+use crate::canister::erc20_transactions::{
+    approve, burn_as_owner, burn_own_tokens, mint_as_owner, mint_test_token, transfer,
+    transfer_from,
+};
+use crate::canister::is20_auction::{
+    auction_info, bid_cycles, bidding_info, run_auction, AuctionError, BiddingInfo,
+};
+use crate::canister::is20_notify::{approve_and_notify, consume_notification, notify};
+use crate::canister::is20_transactions::{batch_transfer, transfer_include_fee};
+use crate::canister::TokenCanister;
+use crate::principal::{CheckedPrincipal, Owner};
+use crate::state::CanisterState;
+use crate::types::{
+    AuctionInfo, Metadata, StatsData, TokenInfo, TxError, TxId, TxReceipt, TxRecord,
+};
+
+pub type AsyncReturn<'a, T> = Pin<Box<dyn Future<Output = T> + 'a>>;
```

Comment:
Why not importing it from canister-sdk directly?

Reply:
Yeah, I haven't thought about it. On it